### PR TITLE
Wlt 695 deposit transfer

### PR DESCRIPTION
**Pull request by pol9kov**
**- What I did**
Added deposit.transfer function
**- How I did it**
Added deposit.transfer method in contract deposit 
**- How to verify it**
Launch deposit_transfer_test.go
**- Description for the changelog**

added deposit.transfer

### CODE REVIEW

File: logicrunner/builtin/contract/deposit/deposit.go [hunk 1]
```diff
@@ -21,14 +21,17 @@ import (
...
 
 	"github.com/insolar/insolar/logicrunner/goplugin/foundation"
 )
 
 type status string
 
 const (
-	month = 30 * 24 * 60 * 60
+	// month = 30 * 24 * 60 * 60
+	month = 0
```
Comment (Gorlyn) [inline logicrunner/builtin/contract/deposit/deposit.go]: Maybe use one pulse for testing?

File: functest/deposit_transfer_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+func TestDepositTransferAnotherTx(t *testing.T) {
+	member := fullMigration(t, "Eth_TxHash_test")
+
+	_, err := signedRequest(member, "deposit.transfer", map[string]interface{}{"amount": "100", "ethTxHash": "Eth_TxHash_foo"})
+	require.Contains(t, err.Error(), "can't find deposit")
+}
+
+func TestDepositTransferThashAmount(t *testing.T) {
```
Comment (megge-dream) [inline functest/deposit_transfer_test.go]: Its better to call it `TestDepositTransferWrongValueAmount`

File: functest/deposit_transfer_test.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDepositTransferToken(t *testing.T) {
```
Comment (megge-dream) [inline functest/deposit_transfer_test.go]: What this test checks? Trying to do transfer 10 times and check user's balance after?

Comment (pol9kov) [reply functest/deposit_transfer_test.go]: It waits hold period. You can transfer only after hold period. Hold period is 6 months but I set it to 1 pulse

File: functest/deposit_transfer_test.go (hunk 2 above)
Comment (pol9kov) [reply functest/deposit_transfer_test.go]: Okey)

File: logicrunner/builtin/contract/deposit/deposit.go [hunk 4]
```diff
@@ -126,3 +129,66 @@ func (d *Deposit) Confirm(migrationDaemonIndex int, migrationDaemonRef string, t
...
+		return fmt.Errorf("failed to get pulse number: %s", err.Error())
+	}
+	if d.PulseDepositUnHold > p {
+		return fmt.Errorf("hold period didn't end")
+	}
+
+	return nil
+}
+
+// MapMarshal gets deposit information.
```
[the remark this answers is not in the document]
Comment (pol9kov) [reply logicrunner/builtin/contract/deposit/deposit.go]: Thank you

File: logicrunner/builtin/contract/deposit/deposit.go (hunk 1 above)
Comment (pol9kov) [reply logicrunner/builtin/contract/deposit/deposit.go]: Done

**Pull request merged**